aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -55,5 +55,161 @@ Tensor _s_where_cpu(const Tensor& condition, const Tensor& self, const Tensor& o
   return ret;
 }
 
+template <typename scalar>
+struct LeOp {
+  static void apply(Tensor& ret, const Tensor& self, Scalar other) {
+    auto other_val = other.to<scalar>();
+    CPU_tensor_apply2<uint8_t, scalar>(ret, self,
+        [other_val](uint8_t& ret_val, scalar self_val) {
+        ret_val = self_val <= other_val;
+      }
+    );
+  }
+};
+
+template <typename scalar>
+struct GeOp {
+  static void apply(Tensor& ret, const Tensor& self, Scalar other) {
+    auto other_val = other.to<scalar>();
+    CPU_tensor_apply2<uint8_t, scalar>(ret, self,
+        [other_val](uint8_t& ret_val, scalar self_val) {
+        ret_val = self_val >= other_val;
+      }
+    );
+  }
+};
+
+template <typename scalar>
+struct EqOp {
+  static void apply(Tensor& ret, const Tensor& self, Scalar other) {
+    auto other_val = other.to<scalar>();
+    CPU_tensor_apply2<uint8_t, scalar>(ret, self,
+        [other_val](uint8_t& ret_val, scalar self_val) {
+        ret_val = self_val == other_val;
+      }
+    );
+  }
+};
+
+template <typename scalar>
+struct NeOp {
+  static void apply(Tensor& ret, const Tensor& self, Scalar other) {
+    auto other_val = other.to<scalar>();
+    CPU_tensor_apply2<uint8_t, scalar>(ret, self,
+        [other_val](uint8_t& ret_val, scalar self_val) {
+        ret_val = self_val != other_val;
+      }
+    );
+  }
+};
+
+template <typename scalar>
+struct LeOpDouble {
+  static void apply(Tensor& result, const Tensor& self, Scalar other) {
+    auto other_val = other.to<double>();
+    CPU_tensor_apply2<uint8_t, scalar>(result, self,
+        [other_val](uint8_t& result_val, scalar self_val) {
+        result_val = convert<scalar, double>(self_val) <= other_val;
+      }
+    );
+  }
+};
+
+template <typename scalar>
+struct GeOpDouble {
+  static void apply(Tensor& result, const Tensor& self, Scalar other) {
+    auto other_val = other.to<double>();
+    CPU_tensor_apply2<uint8_t, scalar>(result, self,
+        [other_val](uint8_t& result_val, scalar self_val) {
+        result_val = convert<scalar, double>(self_val) >= other_val;
+      }
+    );
+  }
+};
+
+template <typename scalar>
+struct EqOpDouble {
+  static void apply(Tensor& result, const Tensor& self, Scalar other) {
+    auto other_val = other.to<double>();
+    CPU_tensor_apply2<uint8_t, scalar>(result, self,
+        [other_val](uint8_t& result_val, scalar self_val) {
+        result_val = convert<scalar, double>(self_val) == other_val;
+      }
+    );
+  }
+};
+
+template <typename scalar>
+struct NeOpDouble {
+  static void apply(Tensor& result, const Tensor& self, Scalar other) {
+    auto other_val = other.to<double>();
+    CPU_tensor_apply2<uint8_t, scalar>(result, self,
+        [other_val](uint8_t& result_val, scalar self_val) {
+        result_val = convert<scalar, double>(self_val) != other_val;
+      }
+    );
+  }
+};
```

Comment:
We should be using templates for those ops. They all differ in a single line.

Reply:
@apaszke I couldn't think of a way to do this with templating. If I'm missing something, please let me know. THC also defines these ops separately: https://github.com/pytorch/pytorch/blob/502aaf39cf4a878f9e4f849e5f409573aa598aa9/aten/src/THC/THCTensorMathCompare.cuh

I think it's doable with a macro but I don't know if I should use one here.

Reply:
[Here's an example that uses templates to implement all of them](https://godbolt.org/g/skczKd).